src/common/cockpitpipe.c

```diff
@@ -1116,11 +1116,18 @@ cockpit_pipe_connect (const gchar *name,
   else
     {
       if (!g_unix_set_fd_nonblocking (sock, TRUE, NULL))
-        g_return_val_if_reached (NULL);
+        {
+          close (sock);
+          g_return_val_if_reached (NULL);
```

Comment:
Out of interest: How would that be related to having to close the fd? `g_return_()` is not like `abort()`, so memory or fd leaks do matter here.